new-docs/source/tutorial/4-python.ipynb

```diff
@@ -0,0 +1,260 @@
+    "argument can be provided to `python.define`\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@python.define(outputs=[\"c\", \"d\"])\n",
+    "def NamedOutputTaskDef(a, b):\n",
+    "    \"\"\"Sample function for testing\"\"\"\n",
+    "    return a + b, a - b\n",
+    "\n",
+    "named_output_task = NamedOutputTaskDef(a=2, b=1)\n",
+    "\n",
+    "outputs = named_output_task()\n",
+    "\n",
+    "print(outputs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The input and output field attributes automatically extracted from the function, explicit\n",
+    "attributes can be augmented"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@python.define(\n",
+    "    inputs={\"a\": python.arg(allowed_values=[1, 2, 3]), \"b\": python.arg(default=10.0)},\n",
+    "    outputs={\n",
+    "        \"c\": python.out(type=float, help=\"the sum of the inputs\"),\n",
+    "        \"d\": python.out(type=float, help=\"the difference of the inputs\"),\n",
+    "    },\n",
+    ")\n",
+    "def AugmentedTaskDef(a, b):\n",
+    "    \"\"\"Sample function for testing\"\"\"\n",
+    "    return a + b, a - b"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Type annotations\n",
+    "\n",
+    "If provided, type annotations are included in the task definition, and are checked at\n",
+    "the time of parameterisation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pydra.design import python\n",
+    "\n",
+    "# Note that we use CamelCase as the function is translated to a class\n",
+    "\n",
+    "@python.define\n",
+    "def MyTypedTask(a: int, b: float) -> float:\n",
+    "    \"\"\"Sample function for testing\"\"\"\n",
+    "    return a + b\n",
+    "\n",
+    "try:\n",
+    "    # 1.5 is not an integer so this should raise a TypeError\n",
+    "    my_typed_task = MyTypedTask(a=1.5, b=2.0)\n",
+    "except TypeError as e:\n",
+    "    print(f\"Type error caught: {e}\")\n",
+    "else:\n",
+    "    assert False, \"Expected a TypeError\"\n",
+    "\n",
+    "# While 2 is an integer, it can be implicitly coerced to a float\n",
+    "my_typed_task = MyTypedTask(a=1, b=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Docstring parsing\n",
+    "\n",
+    "Instead of explicitly providing help strings and output names in `inputs` and `outputs`\n",
+    "arguments, if the function describes the its inputs and/or outputs in the doc string, \n",
+    "in either reST, Google or NumpyDoc style, then they will be extracted and included in the\n",
+    "input or output fields\n"
```

Comment:
the output doesn't have `c` and `d`, just `out` and no `help`. Not sure if this is by design, or it is not working properly. If it's by design, perhaps it could be explained

Reply:
no, that is a bug. Will have a look

Reply:
yes, I saw the tests, and they seem to work well, but they do have explicit name of output in the decorator, i.e., `@python.define(outputs=["c", "d"])`